reinforcement_learning/bindings/cs/rl.net.native/rl.net.ranking_response.cc

```diff
@@ -0,0 +1,79 @@
+#include "rl.net.ranking_response.h"
+
+class ranking_enumerator_adapter
+{
+private:
+    r::ranking_response::ranking_iterator current;
+    r::ranking_response::ranking_iterator end;
+
+public:
+    inline ranking_enumerator_adapter(r::ranking_response* response) : current{response->begin()}, end{response->end()}
+    {}
+
+public:
+    inline int move_next()
+    {
+        if ((this->current != this->end) &&
```

Comment:
I am wondering why you need 2 conditions here. Cannot be done with one test only?

Reply:
We only overload the "not equals" operator for this ranking iterator. Given that we need to decide how we want to deal with unsigned integer types in the C# bindings, I ended up not pushing this part through yet.